chaoscenter/web/src/views/CreateNewUser/CreateNewUser.tsx

```diff
@@ -0,0 +1,144 @@
+import type { UseMutateFunction } from '@tanstack/react-query';
+import React from 'react';
+import { Color, FontVariation } from '@harnessio/design-system';
+import { Layout, Container, FormInput, ButtonVariation, Text, Button } from '@harnessio/uicore';
+import { Formik, Form, FormikProps } from 'formik';
+import { Icon } from '@harnessio/icons';
+import * as Yup from 'yup';
+import type { CreateUserMutationProps, User } from '@api/auth/index.ts';
+import { useStrings } from '@strings';
+
+interface CreateNewUserViewProps {
+  createNewUserMutation: UseMutateFunction<User, unknown, CreateUserMutationProps<never>, unknown>;
+  createNewUserMutationLoading: boolean;
+  handleClose: () => void;
+}
+
+interface CreateNewUserFormProps {
+  name: string;
+  email: string;
+  username: string;
+  password: string;
+  reEnterPassword: string;
+}
+
+export default function CreateNewUserView(props: CreateNewUserViewProps): React.ReactElement {
+  const { createNewUserMutation, createNewUserMutationLoading, handleClose } = props;
+  const { getString } = useStrings();
+
+  function handleSubmit(values: CreateNewUserFormProps): void {
+    createNewUserMutation(
+      {
+        body: {
+          name: values.name,
+          email: values.email,
+          username: values.username,
+          password: values.password,
+          role: 'user'
+        }
+      },
+      {
+        onSuccess: () => {
+          handleClose();
+        }
+      }
+    );
+  }
+  function isSubmitButtonDisabled(formikProps: FormikProps<CreateNewUserFormProps>): boolean {
+    return (
+      formikProps.values.name.length === 0 ||
+      formikProps.values.email.length === 0 ||
+      formikProps.values.username.length === 0 ||
+      formikProps.values.password.length === 0 ||
+      formikProps.values.reEnterPassword.length === 0 ||
+      formikProps.errors.name !== undefined ||
+      formikProps.errors.email !== undefined ||
+      formikProps.errors.username !== undefined ||
+      formikProps.errors.password !== undefined ||
+      formikProps.errors.reEnterPassword !== undefined ||
+      formikProps.values.password !== formikProps.values.reEnterPassword
+    );
+  }
```

Comment:
remove this

Reply:
removed, missed out on adding validation. Updated the form